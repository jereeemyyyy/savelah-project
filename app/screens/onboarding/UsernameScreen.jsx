import { View, Text, TextInput, TouchableOpacity } from 'react-native';
import { useState, useEffect } from 'react';
import { useNavigation } from "@react-navigation/native";
import { supabase } from '../../../lib/supabase';

export default function UsernameScreen() {
    const [username, setUsername] = useState('');
    const navigation = useNavigation();
    const [userId, setUserId] = useState(null);

    // Retrieve the current user's information
    useEffect(() => {
        const fetchUser = async () => {
            try {
                const { data: { user } } = await supabase.auth.getUser();
                if (user) {
                    setUserId(user.id);
                } else {
                    console.log("No user found");
                }
            } catch (error) {
                console.error("Error fetching user:", error.message);
            }
        };

        fetchUser();
    }, []);


    const handleNext = async () => {
        const user_Id = userId;

        // Call the function to save the username
        const { error } = await saveUsername(user_Id, username);

        if (error) {
            console.error('Error saving username:', error);
        } else {
            navigation.navigate('Photo', { username, userId });
        }
    };


    // Insert the username into the profiles table
    const saveUsername = async (userId, username) => { 
        const { data, error } = await supabase
            .from('profiles')
            .update({ username })
            .eq('id', userId);

        return { data, error };
    };



    return (
        <View className="flex-1 bg-gray-800">
            <Text className="font-bold text-4xl w-72 mt-8 mb-4 mx-4 text-white">Welcome to SaveLah!</Text>
            <Text className="text-lg mx-4 mb-4 text-gray-400">Get started by creating a Username.</Text>

            <View className="flex-1 items-center">
                <TextInput
                    className="p-3 bg-gray-100 rounded mb-4 w-52"
                    value={username}
                    onChangeText={setUsername}
                    placeholder="Enter Username"
                />
                <TouchableOpacity className="py-3 bg-indigo-500 rounded mb-4 w-20 items-center" onPress={handleNext}>
                        <Text className="text-white">Next</Text>
                </TouchableOpacity>
            </View>
            
        </View>
    );
}
