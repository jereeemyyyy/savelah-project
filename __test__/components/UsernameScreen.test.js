import React from 'react';
import renderer from 'react-test-renderer';
import UsernameScreen from '../../app/screens/onboarding/UsernameScreen';

jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native');
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: jest.fn(),
        }),
    };
});

describe('UsernameScreen', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<UsernameScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the expected text elements', () => {
        const component = renderer.create(<UsernameScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'welcomeText' }).props.children).toBe('Welcome to SaveLah!');
        expect(instance.findByProps({ testID: 'introText' }).props.children).toBe('Get started by creating a Username.');
    });

    it('contains the username input component', () => {
        const component = renderer.create(<UsernameScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'usernameInput' })).toBeTruthy();
    });

    it('contains the Next button', () => {
        const component = renderer.create(<UsernameScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'nextButton' })).toBeTruthy();
    });
});
