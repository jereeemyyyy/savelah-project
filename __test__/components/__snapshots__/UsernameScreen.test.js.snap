// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsernameScreen renders correctly 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1f2937",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontWeight": "700",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "width": 288,
        },
        {
          "marginTop": 32,
        },
        {
          "marginBottom": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "color": "#fff",
        },
      ]
    }
    testID="welcomeText"
  >
    Welcome to SaveLah!
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 18,
          "lineHeight": 28,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "marginBottom": 16,
        },
        {
          "color": "#9ca3af",
        },
      ]
    }
    testID="introText"
  >
    Get started by creating a Username.
  </Text>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Username"
      style={
        [
          {
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          },
          {
            "backgroundColor": "#f3f4f6",
          },
          {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          {
            "marginBottom": 16,
          },
          {
            "width": 208,
          },
        ]
      }
      testID="usernameInput"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#6366f1",
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "marginBottom": 16,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingTop": 12,
          "width": 80,
        }
      }
      testID="nextButton"
    >
      <Text
        style={
          [
            {
              "color": "#fff",
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
